### PR TITLE
Driving a course in Metadrive test 

### PR DESCRIPTION
**Pull request by bongbui321**
For [#30694](https://github.com/commaai/openpilot/issues/30694) (WIP). This is a draft for the bounty. Currently has 3 failure cases: 2 collisions and drive out of lane. I have a few questions regarding the bounty: 

1. Do you want a new driving course from the old one I'm not sure what you mean by driving a defined course?
2. For logging as artifacts, this would make this bounty the same as [#30693](https://github.com/commaai/openpilot/issues/30693) right? since you would have to be able to run on CI anyway, so do you want that part to be on the other bounty or this one still?

### CODE REVIEW

**Comment by jnewb1** [comment]
> For logging as artifacts, this would make this bounty the same as #30693 right?

This just means we have to get loggerd and encoderd running on PC and it creating logs, adding it as an artifact would be easy to add once we have it creating the logs

**Comment by jnewb1** [comment]
> Do you want a new driving course from the old one I'm not sure what you mean by driving a defined course?

The circle course is fine, just have it complete that.

**Comment by adeebshihadeh** [comment]
> 2. For logging as artifacts, this would make this bounty the same as [#30693](https://github.com/commaai/openpilot/issues/30693) right? since you would have to be able to run on CI anyway, so do you want that part to be on the other bounty or this one still?

Good point, I moved that to the other bounty.

File: tools/sim/tests/test_sim_bridge.py [hunk 1]
```diff
@@ -62,17 +62,27 @@ def test_engage(self):
...
         if control_active == min_counts_control_active:
           break
-
     self.assertEqual(min_counts_control_active, control_active, f"Simulator did not engage a minimal of {min_counts_control_active} steps was {control_active}")
 
+    # Drive course
+    q.put("reset")
+    start_time = time.monotonic()
+    user_disengage_once = False
+    disengage_events = ('stockAeb', 'fcw', 'ldw')
```
Comment (adeebshihadeh) [inline tools/sim/tests/test_sim_bridge.py:76]: this doesn't tell you what you need to know. the point of the bounty is to determine whether openpilot stayed in its lane, drove a reasonable, etc. and finished the course

**Comment by adeebshihadeh** [comment]
@bongbui321 I'm unlocking the bounty since it's been a couple weeks. feel free to open the PR back up if you start working on it again

**Pull request closed without merging**